packages/vscode-mdx/README.md

```diff
@@ -60,6 +61,113 @@ repository readme.
 Syntax highlighting for MDX is based on the
 [MDX TextMate grammar](https://github.com/wooorm/markdown-tm-language).
 
+### Custom Languages in Code Blocks
+
+MDX for Visual Studio Code supports syntax highlighting for a number of
+well-known languages in code blocks.
+However, it’s impossible to support all languages within the MDX extension.
+Instead, if an extensions adds support for a language, it can add support for
+MDX code blocks.
+
+To support MDX code blocks in your Visual Studio Code extension, use the
+following template.
+Replace `LANGUAGE` with your actual language and remove comments.
+Save the file to `syntaxes/mdx.LANGUAGE.tmLanguage.json`.
+
+````jsonc
+{
+  "fileTypes": [],
+  // This can be something else.
+  "scopeName": "mdx.LANGUAGE.codeblock",
+  "injectionSelector": "L:source.mdx",
+  "patterns": [
+    {
+      // This references the repository key below.
+      "include": "#LANGUAGE-code-block"
```

Comment:
i would use names that are closer to how they are used in markdown-tm-language. I don’t think that uses `code-block`?

Reply:
It doesn’t really matter. I aligned this with other extensions’ names in markdown injections. I’m fine with changing it in this template.

I could also include the patterns from the included repository directly. Then this name is gone, so there’s less that can go wrong replacing, there’s no need for the `repository` field, and no need to explain the names need to match.

```jsonc
{
  "fileTypes": [],
  "scopeName": "mdx.LANGUAGE.codeblock",
  "injectionSelector": "L:source.mdx",
  "patterns": [
    {
      "begin": "(?:^|\\G)[\\t ]*(`{3,})(?:[\\t ]*((?i:(?:.*\\.)?LANGUAGE))(?:[\\t ]+((?:[^\\n\\r`])+))?)(?:[\\t ]*$)",
      "beginCaptures": {
        "1": {
          "name": "string.other.begin.code.fenced.mdx"
        },
        "2": {
          "name": "entity.name.function.mdx"
        }
      },
      "contentName": "meta.embedded.LANGUAGE",
      "end": "(\\1)(?:[\\t ]*$)",
      "endCaptures": {
        "1": {
          "name": "string.other.end.code.fenced.mdx"
        }
      },
      "name": "markup.code.LANGUAGE.mdx",
      "patterns": [
        {
          "include": "source.LANGUAGE"
        }
      ]
    },
    {
      "begin": "(?:^|\\G)[\\t ]*(~{3,})(?:[\\t ]*((?i:(?:.*\\.)?LANGUAGE))(?:[\\t ]+((?:[^\\n\\r])+))?)(?:[\\t ]*$)",
      "beginCaptures": {
        "1": {
          "name": "string.other.begin.code.fenced.mdx"
        },
        "2": {
          "name": "entity.name.function.mdx"
        }
      },
      "contentName": "meta.embedded.LANGUAGE",
      "end": "(\\1)(?:[\\t ]*$)",
      "endCaptures": {
        "1": {
          "name": "string.other.end.code.fenced.mdx"
        }
      },
      "name": "markup.code.LANGUAGE.mdx",
      "patterns": [
        {
          "include": "source.LANGUAGE"
        }
      ]
    }
  ]
}
```

Reply:
right, yes, I prefer that